### PR TITLE
Add recommended content to dialogue match form and restyle

### PR DESCRIPTION
**Pull request by lagerros**
* Add a card to the dialogues match form containing recommended content by the other user (top tags, posts, comments)
* Restyle the whole dialogue match form to look cleaner given this
* Fix some dependency cycles that resulted

Before: 
image

After: 
image

### CODE REVIEW

**Comment by rjmk** [comment]
@lagerros can we do an AB Test here?

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:978]: does this need to be cleaned up?

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:831]: `(allRecommendations && allRecommendations.length > 0)` feels unnecessary given the check at L817

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (b0b3rt) [inline packages/lesswrong/components/users/DialogueMatchingPage.tsx:797]: Is this (and most of the preceeding logic) copied from somewhere?  Looks very familiar

**Review by rjmk — DISMISSED** [review]
We've got a bunch of different ideas for changing this form incoming. I want to AB Test the different changes, including the null change. Could you start the sequence of AB Tests by creating one for this PR?

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx
Comment (b0b3rt) [inline packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:435]: If you're now registering the `CommentView` component and wrapping it in the `styles` HoC, but it previously took `classes` as a prop, that means wherever you were using it before, you were passing in `classes` to it manually.  I don't see any diffs that reflect that having been changed/fixed so I'm a bit suspicious that something is broken here

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx:426]: tried numbers, didn't work for a strange reason, had to use pixel values. can show you if you want

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx:797]: it is. Considered putting it into a reusable hook, but, third times the charm

File: packages/lesswrong/components/users/DialogueMatchingPage.tsx
Comment (lagerros) [reply packages/lesswrong/components/users/DialogueMatchingPage.tsx]: legit

**Review by rjmk — COMMENTED** [review]
Hopefully this removes my request for feedback without creating an approval

File: packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx
Comment (lagerros) [reply packages/lesswrong/components/dialogues/DialogueRecommendationRow.tsx:435]: It's only used in two places (frontpage recommendations, and dialogue match form). I checked behavior in both before making the PR, and it seemed fine. 

Nonetheless, I fixed the issue (properly imported CommentView and removed classes argument), and thanks for catching!

**Comment by rjmk** [comment]
Yeah I'm just keeping an eye and waiting for Robert's approving review

**Review by jpaddison3 — COMMENTED** [review]
No objection

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**
